src/darwin/Framework/CHIP/MTRDefines_Internal.h

```diff
@@ -150,3 +150,7 @@ typedef struct {} variable_hidden_by_mtr_hide;
         }                                                                                                                  \
     }
 #endif
+
+#ifndef YES_NO
```

Comment:
I think it's safer to prefix this as `MTR_YES_NO`. Also no need for the ifdef -- if the macro already has a conflicting definition a warning / error would be better than silently the wrong macro.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/35494